comp/core/workloadmeta/collectors/internal/cloudfoundry/vm/cf_vm_test.go

```diff
@@ -248,10 +253,9 @@ func TestPullNoContainers(t *testing.T) {
 	}
 
 	err := c.Pull(context.TODO())
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
-	evs := workloadmetaStore.GetNotifiedEvents()
-	assert.Empty(t, evs)
+	assert.Empty(t, workloadmetaStore.ListContainers())
```

Comment:
Wouldn't this give a false positive is `ListContainers()` was executed before the `handleEvents` method of workloadmeta?

Reply:
Yes, but the test would fail at least some times if there was an issue.
The alternative is using a fixed sleep time directly or with something like `assert.Never`. Not sure if it's worth it.

Reply:
I see. 
So this makes the test non-deterministic, which is not great. 

How about using `workloadmetaStore.Set` method in order to manually inject a placeholder container entity just after the pull. The assertion can be then changed to assert that `len(workloadmetaStore.ListContainers()) == 1` and that `workloadmetaStore.ListContainers()[0]` is the placeholder container that we injected. 

This will be correct because workloadmeta store handles events synchronously.

Reply:
Changed :+1: